torch/onnx/symbolic_opset9.py

```diff
@@ -3398,7 +3398,8 @@ def Loop(ctx: torch.onnx.SymbolicContext, g, *inputs, **attrs):
             for i, b_in in enumerate(b.inputs()):
                 if i == 0 and i < len(inputs):
                     b_in.setType(inputs[i].type())
-                if i > 0 and (i + 1) < len(inputs):
+                # For optional block inputs, input type could be NoneType, which we don't want to propagate.
+                if i > 0 and (i + 1) < len(inputs) and not isinstance(inputs[i + 1].type(), NoneType):
```

Comment:
what happens for using optional block inputs with loop opset < 15. This code looks like it will be silently not setting types for it. Will this end up with invalid model?